datalad/core/distributed/clone.py

```diff
@@ -585,15 +585,125 @@ def postclonecfg_ria(ds, props):
                    if s.get('annex-externaltype', None) == 'ora'
     ]
     if not ria_remotes:
-        lgr.debug("Found no RIA special remote")
+        # nothing autoenabled. Let's check origin's config for
+        # datalad.ora-remote.uuid as stored by create-sibling-ria and enable try
+        # enabling that one.
+        lgr.debug("Found no autoenabled ORA special remote. Trying to look it"
+                  "up in source config ...")
+
+        # First figure whether we cloned via SSH, HTTP or local path and then
+        # get that config file the same way:
+
+        config_content = None
+        scheme = props['source'].split(':')[0][4:]
+        if scheme == 'http':
+
+            # Note: Following outcommented code was proof how to check via HTTP.
+            #       However, for now we only want to annexenable origin in case
+            #       of HTTP, not the special remote. Hence it's uuid isn't
+            #       actually required. (see gh-4410)
+
+            # import requests
+            # try:
+            #     response = requests.get("{}{}config".format(
+            #         props['giturl'],
+            #         '/' if not props['giturl'].endswith('/') else '')
+            #     )
+            #     config_content = response.text
+            # except requests.RequestException as e:
+            #     lgr.debug("Failed to get config file from source:\n%s",
+            #               exc_str(e))
+
+            # config_content still None should lead to skip everything special
+            # remote related. Instead enable origin to deliver annex keys via
+            # HTTP:
+            lgr.debug("Trying to enable remote origin for git-annex")
+            ds.repo.enable_remote('origin')
+            ds.repo.fsck(remote='origin', fast=True)
+
+        elif scheme == 'ssh':
+
+            # TODO: switch the following to proper command abstraction:
+            from datalad.distributed.ora_remote import SSHRemoteIO, RIARemoteError
+            # SSHRemoteIO ignores the path part ATM. No remote CWD! (To be
+            # changed with command abstractions). So we need to get that part to
+            # have a valid path to origin's config file:
+            from datalad.support.network import URL
+            from datalad.utils import PurePosixPath
+            cfg_path = PurePosixPath(URL(props['giturl']).path) / 'config'
+            op = SSHRemoteIO(props['giturl'])
+            try:
+                # TODO: double-check path
+                response = op.read_file(cfg_path)
+                config_content = response
+            except RIARemoteError as e:
+                lgr.debug("Failed to get config file from source:\n%s",
+                          exc_str(e))
```

Comment:
Just a side-comment: I don't know why we have `exc_str()` (compares to having a `__str__()` in our exceptions), but there is a cost for each *not shown* debug message for using it -- in contrast to just passing `e` and let stringification happen later, if actually needed.

Reply:
`exc_str` is about getting its original location from traceback. TBH I forgot when exactly this is needed and when it's superfluous.

Reply:
It did come super useful to me many times when I had to dig out how the heck an exception came about to be logged (especially is swallowed upstairs somewhere).  We have `DATALAD_EXC_STR_TBLIMIT` env variable to instruct to include such information.

As for performance -- if you are hitting exceptions, I think a few `ms` spent on collecting a bit more data would not really be what causes bad UX ;)

Reply:
<details>
<summary>Did some basic timing of exc_str. Takes ~2ms on a rudimentary use case. IMHO it is worth a "wait".  2ms means that you would wait for 1 sec if you manage to throw and catch 500 exceptions.  Did not time how long logging itself would be taking in that case etc</summary> 

```shell
$> DATALAD_EXC_STR_TBLIMIT=10 python ../trash/bench_exc_str.py
str:     0.000009 = my exception
exc_str: 0.001914 = my exception [bench_exc_str.py:<module>:12,bench_exc_str.py:f1:9,bench_exc_str.py:f2:8,bench_exc_str.py:f3:7]
(dev3) 1 22499.....................................:Tue 28 Apr 2020 12:56:58 PM EDT:.
(git)lena:~datalad/datalad-master[master]git
$> python ../trash/bench_exc_str.py        
str:     0.000008 = my exception
exc_str: 0.001510 = my exception [bench_exc_str.py:f3:7]
(dev3) 1 22500.....................................:Tue 28 Apr 2020 12:57:03 PM EDT:.
(git)lena:~datalad/datalad-master[master]git
$> cat ../trash/bench_exc_str.py
import timeit
from datalad.dochelpers import exc_str

def f1():
    def f2():
        def f3():
            raise IndexError("my exception")
        f3()
    f2()

try:
    f1()
except Exception as e:
    g = {'e': e, 'exc_str': exc_str}
    print("str:     %f = %s" % (timeit.timeit("str(e)", globals=g, number=100), str(e)))
    print("exc_str: %f = %s" % (timeit.timeit("exc_str(e)", globals=g, number=100), exc_str(e)))

```
</details>